CRM/Mailing/BAO/Mailing.php

```diff
@@ -96,59 +96,268 @@ public function __construct() {
+      "CREATE TEMPORARY TABLE $includedTempTablename
+            (contact_id int primary key, $tempColumn int)
+            ENGINE=HEAP"
+    );
+
+    // Criterias to filter recipients that need to be included
+    $includeFilters = array(
+      "$contact.do_not_email = 0",
+      "$contact.is_opt_out = 0",
+      "$contact.is_deceased <> 1",
+      $location_filter,
+      "$email.email IS NOT NULL",
+      "$email.email != ''",
+      'temp.contact_id IS NULL',
+    );
+
+    if ($isSMSmode) {
       $phone = CRM_Core_DAO_Phone::getTableName();
+      CRM_Utils_SQL_Select::from($phone)
+        ->select(" DISTINCT $phone.id as phone_id ")
+        ->join($contact, " INNER JOIN $contact ON $phone.contact_id = $contact.id ")
+        ->join('gc', " INNER JOIN civicrm_group_contact gc ON $contact.id = gc.contact_id ")
+        ->join('mg', " INNER JOIN civicrm_mailing_group mg ON gc.group_id = mg.entity_id AND mg.entity_table = 'civicrm_group' ")
+        ->join('temp', " LEFT JOIN $excludeTempTablename temp ON $contact.id = temp.contact_id ")
+        ->where(array(
+            "mg.group_type = 'Include'",
+            'mg.search_id IS NULL',
+            "gc.status = 'Added'",
+            "$contact.do_not_sms",
+            "$contact.is_opt_out = 0",
+            "$contact.is_deceased <> 1",
+            "$phone.phone_type_id = " . CRM_Utils_Array::value('Mobile', CRM_Core_OptionGroup::values('phone_type', TRUE, FALSE, FALSE, NULL, 'name')),
+            "$phone.phone IS NOT NULL",
+            "$phone.phone != ''",
+            "mg.mailing_id = #mailingID",
+            'temp.contact_id IS null',
+        ))
+        ->param('#mailingID', $mailingID)
+        ->replaceInto($includedTempTablename, array('contact_id', 'phone_id'))
+        ->execute();
+    }
+    else {
+      // Get the group contacts, but only those which are not in the
+      // exclusion temp table.
+      CRM_Utils_SQL_Select::from($email)
+        ->select("$contact.id as contact_id, $email.id as email_id")
+        ->join($contact, " INNER JOIN $contact ON $email.contact_id = $contact.id ")
+        ->join('gc', " INNER JOIN civicrm_group_contact gc ON gc.contact_id = $contact.id ")
+        ->join('mg', " INNER JOIN civicrm_mailing_group mg  ON  gc.group_id = mg.entity_id AND mg.search_id IS NULL ")
+        ->join('temp', " LEFT JOIN $excludeTempTablename temp ON $contact.id = temp.contact_id ")
+        ->where('gc.group_id IN (#groups) AND gc.status = "Added"')
+        ->where($includeFilters)
+        ->groupBy(array("$contact.id", "$email.id"))
+        ->replaceInto($includedTempTablename, array('contact_id', 'email_id'))
+        ->param('#groups', $recipientsGroup['Include'])
+        ->execute();
+    }
+
+    if (count($includeSmartGroupIDs)) {
+      CRM_Utils_SQL_Select::from($contact)
+        ->select("$contact.id as contact_id, $email.id as email_id")
+        ->join($email, " INNER JOIN $email ON $email.contact_id = $contact.id ")
```

Comment:
Hi @monishdeb I've just spotted something - I think these changes will break trying to include a smart group in an SMS type mailing? There is no if statement, so this will insert email IDs into the 'phone ID' column in the temporary table?

I haven't proved this definitively yet, working on setting up my localhost to test.

Reply:
Thanks, @JKingsnorth for notifying this break. I am going to a create a separate PR for this fix.